netty/src/main/java/io/grpc/transport/netty/NettyClientHandler.java

```diff
@@ -150,18 +153,27 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
    */
   @Override
   public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) {
-    if (msg instanceof CreateStreamCommand) {
-      createStream((CreateStreamCommand) msg, promise);
-    } else if (msg instanceof SendGrpcFrameCommand) {
-      sendGrpcFrame(ctx, (SendGrpcFrameCommand) msg, promise);
-    } else if (msg instanceof CancelStreamCommand) {
-      cancelStream(ctx, (CancelStreamCommand) msg, promise);
-    } else if (msg instanceof RequestMessagesCommand) {
-      ((RequestMessagesCommand) msg).requestMessages();
-    } else if (msg instanceof SendPingCommand) {
-      sendPingFrame(ctx, (SendPingCommand) msg, promise);
-    } else {
-      throw new AssertionError("Write called for unexpected type: " + msg.getClass().getName());
+    try {
+      if (msg instanceof CreateStreamCommand) {
+        createStream((CreateStreamCommand) msg, promise);
+      } else if (msg instanceof SendGrpcFrameCommand) {
+        sendGrpcFrame(ctx, (SendGrpcFrameCommand) msg, promise);
+      } else if (msg instanceof CancelStreamCommand) {
+        cancelStream(ctx, (CancelStreamCommand) msg, promise);
+      } else if (msg instanceof RequestMessagesCommand) {
+        ((RequestMessagesCommand) msg).requestMessages();
+      } else if (msg instanceof SendPingCommand) {
+        sendPingFrame(ctx, (SendPingCommand) msg, promise);
+      } else {
+        throw new AssertionError("Write called for unexpected type: " + msg.getClass().getName());
+      }
+    } catch (StatusException e) {
+      promise.setFailure(e);
+      try {
+        super.close(ctx, ctx.newPromise());
+      } catch (Exception ce) {
```

Comment:
This isn't what I was encouraging when I was suggesting to add "throws Exception" to `createStream()` and `write()`.

Instead, I was suggesting that this `} catch (Exception ce) {` doesn't need to be here at all, because `write()` can `throws Exception`. I do think the super.close() needs to be back where it was in `createStream()`.

The changes for propagating StatusException further seem like a step backward. I liked it more with how it was before.